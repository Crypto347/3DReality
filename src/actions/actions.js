import {
    TEST_TYPE
} from '../constants/actionTypes';

export function addArticle(payload){
    return {
        type: TEST_TYPE,
        payload
    }
};
