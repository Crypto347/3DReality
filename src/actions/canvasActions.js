import * as actionTypes from '../constants/actionTypes';

export function moveCircleXCoordinate(x) {
    return { 
        type: actionTypes.MOVE_CIRCLE_X_COORDINATE,
        dx: x
    };
};

export function moveCircleYCoordinate(y) {
    return { 
        type: actionTypes.MOVE_CIRCLE_Y_COORDINATE,
        dy: y
    };
};

export function changeDirectionOfXMove() {
    return { 
        type: actionTypes.CHANGE_DIRECTION_OF_X_MOVE
    };
};

export function changeDirectionOfYMove() {
    return { 
        type: actionTypes.CHANGE_DIRECTION_OF_Y_MOVE
    };
};

