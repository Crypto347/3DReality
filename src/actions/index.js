export {
    checkButtons,
    makePaperclip,
    sellPaperclips,
    updateFunds,
    startUpdatingUnsoldInventory,
    updateUnsoldInventory,
    lowerPrice,
    raisePrice,
    updatePublicDemand,
    toggleMarketingButton,
    marketing,
    marketingNextLevel,
    updateMaxPublicDemand,
    buyWire,
    randomWirePrice,
    toggleWireButton,
    checkExistenceOfWire,
    stop,
    startBuyingWire,
    autoPaperclips,
    autoPaperclipsStart,
    autoClippersAddOne,
    setAutoClipperInitPrice,
    toggleAutoClippersButton,
    toggleMegaClippersButton,
    trustPlusOne,
    increaseOps,
    startDecreasingOps,
    stopDecreasingOps,
    decreaseOps,
    increaseProcessors,
    increaseProcessorsMemory,
    increaseCreativity,
    initProjects,
    checkCardValidity,
    deleteCard,
    showRevTracker,
    addProject,
    removePriceOfProjectOps,
    removePriceOfProjectCreat,
    removePriceOfProjectOpsAndCreat,
    removePriceOfProjectTrust,
    improveAutoClippers, 
    toggleMakePaperclipButton,
    improveWireExtrusion,
    startCreativityCounter,
    creativityTurnOn,
    trustPlusOneFromProject,
    wireExists,
    addLexicalProcessing,
    addCombinatoryHarmonics,
    addTheHadwingerProblem,
    addTheTothSausageConjecture,
    addDonkeySpace,
    improveMarketing,
    showInvestEngine,
    showStrategicModeling,
    sendCommentToTerminal,
    addNewStrategy,
    toggleDropdownInvestments,
    toggleDropdownStrategicModeling,
    closeDropdowns,
    showQuantCompMessage,
    showQuantumComputing,
    changeToQOps,
    startAddingQOps,
    addQOps,
    stopAddingQOps,
    startSubtractingQOps,
    subtractQOps,
    stopSubtractingQOps,
    captureCurrentQOps,
    addChip,
    toggleChip,
    updateOps,
    startDecreasingOperations,
    startTimer,
    clickWireButton,
    showAutoWireBuyer,
    toggleWireBuyerProject,
    autoWireBuyer,
    toggleAutoWireBuyer,
    addMegaClippers,
    showMegaClippers,
    showAutoClippers,
    calcDelayUnsoldInventary,
    stopUpdatingUnsoldInventory,
    startImprovingMarketing,
    updateClipsPerSec,
    startInvestmentsDeposit,
    getDeposit,
    startUpdatingScreen,
    stopUpdatingScreen,
    addInvestmentsLine,
    startUpdatingInvestmentLines,
    stopUpdatingInvestmentLines,
    updateInvestmentsLines,
    startCountingRisk,
    updateInvestmentsTotal,
    updateInvestmentsCash,
    updateInvestmentsStocks,
    startApplyingProfitLoss,
    updateFakeInvestmentsCash,
    startInvestmentsWithdraw,
    updateFundsWithdraw,
    chooseFromDropdown,
    addChosenFromDropdown,
    updateInvestmentsDelay,
    startAddingEmptyInvestmentsLine,
    stopAddingEmptyInvestmentsLine,
    updateAvgRevPerSec,
    updateAvgClipsSoldPerSec,
    megaClippersButtonPressed,
    startMegaClippers,
    improveMegaClippers,
    switchOffOrOnAutoAndMegaClippers,
    startNewTournament
} from './businessActions';
