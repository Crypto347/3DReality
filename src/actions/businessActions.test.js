import configureStore from 'redux-mock-store';
import * as Actions from '../actions';
import * as actionTypes from "../constants/actionTypes";
import { iterator } from 'rxjs/internal-compatibility';

const mockStore = configureStore();
const store = mockStore();

describe('businessActions', () => {
    beforeEach(() => { // Runs before each test in the suite
      store.clearActions();
    });
  
    it('Dispatches the correct action and payload (makePaperclip)', () => {
        const expectedActions = [
            {
                type: actionTypes.MAKE_PAPERCLIP
            }
        ];

        store.dispatch(Actions.makePaperclip());
        expect(store.getActions()).toEqual(expectedActions);
        expect(store.getActions()).toMatchSnapshot();
    })

    it('Dispatches the correct action and payload (updateFunds)', () => {
        const expectedActions = [
            {
                type: actionTypes.UPDATE_FUNDS,
                value: 5
            }
        ];

        store.dispatch(Actions.updateFunds(5));
        expect(store.getActions()).toEqual(expectedActions);
        expect(store.getActions()).toMatchSnapshot();
    })

    it('Dispatches the correct action and payload (checkButtons)', () => {
        const expectedActions = [
            {
                type: actionTypes.CHECK_BUTTONS
            }
        ];

        store.dispatch(Actions.checkButtons());
        expect(store.getActions()).toEqual(expectedActions);
        expect(store.getActions()).toMatchSnapshot();
    })

    it('Dispatches the correct action and payload (sellPaperclips)', () => {
        const expectedActions = [
            {
                type: actionTypes.START_SELLING
            }
        ];

        store.dispatch(Actions.sellPaperclips());
        expect(store.getActions()).toEqual(expectedActions);
        expect(store.getActions()).toMatchSnapshot();
    })

    it('Dispatches the correct action and payload (startUpdatingUnsoldInventory)', () => {
        const expectedActions = [
            {
                type: actionTypes.START_UPDATING_UNSOLD_INVENTORY
            }
        ];

        store.dispatch(Actions.startUpdatingUnsoldInventory());
        expect(store.getActions()).toEqual(expectedActions);
        expect(store.getActions()).toMatchSnapshot();
    })

    it('Dispatches the correct action and payload (stopUpdatingUnsoldInventory)', () => {
        const expectedActions = [
            {
                type: actionTypes.STOP_UPDATING_UNSOLD_INVENTORY
            }
        ];

        store.dispatch(Actions.stopUpdatingUnsoldInventory());
        expect(store.getActions()).toEqual(expectedActions);
        expect(store.getActions()).toMatchSnapshot();
    })

    it('Dispatches the correct action and payload (updateUnsoldInventory)', () => {
        const expectedActions = [
            {
                type: actionTypes.UPDATE_UNSOLD_INVENTORY
            }
        ];

        store.dispatch(Actions.updateUnsoldInventory());
        expect(store.getActions()).toEqual(expectedActions);
        expect(store.getActions()).toMatchSnapshot();
    })

    it('Dispatches the correct action and payload (lowerPrice)', () => {
        const expectedActions = [
            {
                type: actionTypes.LOWER_PRICE
            }
        ];

        store.dispatch(Actions.lowerPrice());
        expect(store.getActions()).toEqual(expectedActions);
        expect(store.getActions()).toMatchSnapshot();
    })

    it('Dispatches the correct action and payload (raisePrice)', () => {
        const expectedActions = [
            {
                type: actionTypes.RAISE_PRICE
            }
        ];

        store.dispatch(Actions.raisePrice());
        expect(store.getActions()).toEqual(expectedActions);
        expect(store.getActions()).toMatchSnapshot();
    })

    it('Dispatches the correct action and payload (updatePublicDemand)', () => {
        const expectedActions = [
            {
                type: actionTypes.UPDATE_PUBLIC_DEMAND
            }
        ];

        store.dispatch(Actions.updatePublicDemand());
        expect(store.getActions()).toEqual(expectedActions);
        expect(store.getActions()).toMatchSnapshot();
    })

    it('Dispatches the correct action and payload (toggleMarketingButton)', () => {
        const expectedActions = [
            {
                type: actionTypes.TOGGLE_MARKETING_BUTTON
            }
        ];

        store.dispatch(Actions.toggleMarketingButton());
        expect(store.getActions()).toEqual(expectedActions);
        expect(store.getActions()).toMatchSnapshot();
    })
    
    it('Dispatches the correct action and payload (marketing)', () => {
        const expectedActions = [
            {
                type: actionTypes.MARKETING
            }
        ];

        store.dispatch(Actions.marketing());
        expect(store.getActions()).toEqual(expectedActions);
        expect(store.getActions()).toMatchSnapshot();
    })

    it('Dispatches the correct action and payload (marketingNextLevel)', () => {
        const expectedActions = [
            {
                type: actionTypes.MARKETING_NEXT_LEVEL
            }
        ];

        store.dispatch(Actions.marketingNextLevel());
        expect(store.getActions()).toEqual(expectedActions);
        expect(store.getActions()).toMatchSnapshot();
    })
  });