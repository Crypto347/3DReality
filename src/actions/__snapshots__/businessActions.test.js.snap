// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`businessActions Dispatches the correct action and payload (addChip) 1`] = `
Array [
  Object {
    "obj": Object {
      "c": 8,
    },
    "type": "ADD_CHIP",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (addChosenFromDropdown) 1`] = `
Array [
  Object {
    "chosen": "I am chosen",
    "index": 0,
    "type": "ADD_CHOSEN_FROM_DROPDOWN",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (addInvestmentsLine) 1`] = `
Array [
  Object {
    "notEmpty": false,
    "obj": "",
    "type": "ADD_INVESTMENTS_LINE",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (addMegaClippers) 1`] = `
Array [
  Object {
    "type": "ADD_MEGA_CLIPPERS",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (addNewStrategy) 1`] = `
Array [
  Object {
    "strategy": "New Strategy",
    "type": "ADD_NEW_STRATEGY",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (addProject) 1`] = `
Array [
  Object {
    "project": Object {
      "b": 7,
    },
    "type": "ADD_PROJECT",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (addQOps) 1`] = `
Array [
  Object {
    "chipsNumber": "chipX",
    "type": "ADD_Q_OPS",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (autoClippersAddOne) 1`] = `
Array [
  Object {
    "type": "AUTO_CLIPPERS_ADD_ONE",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (autoPaperclips) 1`] = `
Array [
  Object {
    "type": "AUTO_PAPERCLIPS",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (autoPaperclipsStart) 1`] = `
Array [
  Object {
    "type": "AUTO_PAPERCLIPS_START",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (autoWireBuyer) 1`] = `
Array [
  Object {
    "type": "AUTO_WIRE_BUYER",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (buyWire) 1`] = `
Array [
  Object {
    "type": "BUY_WIRE",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (calcDelayUnsoldInventary) 1`] = `
Array [
  Object {
    "type": "CALC_DELAY_UNSOLD_INVENTARY",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (captureCurrentQOps) 1`] = `
Array [
  Object {
    "type": "CAPTURE_CURRENT_Q_OPS",
    "val": 276,
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (changeToQOps) 1`] = `
Array [
  Object {
    "type": "CHANGE_TO_Q_OPS",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (checkButtons) 1`] = `
Array [
  Object {
    "type": "CHECK_BUTTONS",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (checkCardValidity) 1`] = `
Array [
  Object {
    "cardId": "card1",
    "i": 0,
    "type": "CHECK_CARD_VALIDITY",
    "valid": true,
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (checkExistenceOfWire) 1`] = `
Array [
  Object {
    "type": "CHECK_EXISTENCE_OF_WIRE",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (chooseFromDropdown) 1`] = `
Array [
  Object {
    "chosen": "I am chosen",
    "index": 1,
    "type": "CHOOSE_FROM_DROPDOWN",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (clearChosenFromStrategicModelingDropdownList) 1`] = `
Array [
  Object {
    "type": "CLEAR_CHOSEN_FROM_STRATEGIC_MODELING_DROPDOWN_LIST",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (clickWireButton) 1`] = `
Array [
  Object {
    "type": "CLICK_WIRE_BUTTON",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (closeDropdowns) 1`] = `
Array [
  Object {
    "type": "CLOSE_DROPDOWNS",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (creativityTurnOn) 1`] = `
Array [
  Object {
    "type": "CREATIVITY_TURN_ON",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (decreaseOps) 1`] = `
Array [
  Object {
    "type": "DECREASE_OPS",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (deleteCard) 1`] = `
Array [
  Object {
    "cardId": "card3",
    "type": "DELETE_CARD",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (getDeposit) 1`] = `
Array [
  Object {
    "type": "GET_DEPOSIT",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (improveAutoClippers) 1`] = `
Array [
  Object {
    "type": "IMPROVE_AUTO_PAPER_CLIPPER",
    "val": 17,
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (improveMarketing) 1`] = `
Array [
  Object {
    "type": "IMPROVE_MARKETING",
    "val": 46,
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (improveMegaClippers) 1`] = `
Array [
  Object {
    "type": "IMPROVE_MEGA_CLIPPERS",
    "val": 36,
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (improveWireExtrusion) 1`] = `
Array [
  Object {
    "type": "IMPROVE_WIRE_EXTRUSION",
    "val": 36,
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (increaseCreativity) 1`] = `
Array [
  Object {
    "type": "INCREASE_CREATIVITY",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (increaseOps) 1`] = `
Array [
  Object {
    "type": "INCREASE_OPS",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (increaseProcessors) 1`] = `
Array [
  Object {
    "type": "INCREASE_PROCESSORS",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (increaseProcessorsMemory) 1`] = `
Array [
  Object {
    "type": "INCREASE_PROCESSORS_MEMORY",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (initProjects) 1`] = `
Array [
  Object {
    "card1": Object {
      "a": 1,
    },
    "card2": Object {
      "a": 2,
    },
    "card3": Object {
      "a": 3,
    },
    "type": "INIT_PROJECTS",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (lowerPrice) 1`] = `
Array [
  Object {
    "type": "LOWER_PRICE",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (makePaperclip) 1`] = `
Array [
  Object {
    "type": "MAKE_PAPERCLIP",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (marketing) 1`] = `
Array [
  Object {
    "type": "MARKETING",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (marketingNextLevel) 1`] = `
Array [
  Object {
    "type": "MARKETING_NEXT_LEVEL",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (megaClippersButtonPressed) 1`] = `
Array [
  Object {
    "type": "MEGA_CLIPPERS_BUTTON_PRESSED",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (raisePrice) 1`] = `
Array [
  Object {
    "type": "RAISE_PRICE",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (randomWirePrice) 1`] = `
Array [
  Object {
    "type": "RANDOM_WIRE_PRICE",
    "value": 15,
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (removePriceOfProjectCreat) 1`] = `
Array [
  Object {
    "creativity": 73,
    "type": "REMOVE_PRICE_OF_PROJECT_CREAT",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (removePriceOfProjectOps) 1`] = `
Array [
  Object {
    "ops": 47,
    "type": "REMOVE_PRICE_OF_PROJECT_OPS",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (removePriceOfProjectOpsAndCreat) 1`] = `
Array [
  Object {
    "creativity": 25,
    "ops": 74,
    "type": "REMOVE_PRICE_OF_PROJECT_OPS_AND_CREAT",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (removePriceOfProjectTrust) 1`] = `
Array [
  Object {
    "trust": 67,
    "type": "REMOVE_PRICE_OF_PROJECT_TRUST",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (sellPaperclips) 1`] = `
Array [
  Object {
    "type": "START_SELLING",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (sendCommentToTerminal) 1`] = `
Array [
  Object {
    "comment": "Testing actions",
    "type": "SEND_COMMENT_TO_TERMINAL",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (setAutoClipperInitPrice) 1`] = `
Array [
  Object {
    "type": "SET_AUTO_CLIPPER_INIT_PRICE",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (showAutoClippers) 1`] = `
Array [
  Object {
    "type": "SHOW_AUTO_CLIPPERS",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (showAutoWireBuyer) 1`] = `
Array [
  Object {
    "type": "SHOW_AUTO_WIRE_BUYER",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (showInvestEngine) 1`] = `
Array [
  Object {
    "type": "SHOW_INVESTMENT_ENGINE",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (showMegaClippers) 1`] = `
Array [
  Object {
    "type": "SHOW_MEGA_CLIPPERS",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (showQuantCompMessage) 1`] = `
Array [
  Object {
    "type": "SHOW_QUANT_COMP_MESSAGE",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (showQuantumComputing) 1`] = `
Array [
  Object {
    "type": "SHOW_QUANTUM_COMPUTING",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (showRevTracker) 1`] = `
Array [
  Object {
    "price": 45,
    "type": "SHOW_REV_TRACKER",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (showRoundAndPlayers) 1`] = `
Array [
  Object {
    "type": "SHOW_ROUND_AND_PLAYERS",
    "val": true,
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (showStrategicModeling) 1`] = `
Array [
  Object {
    "type": "SHOW_STRATEGIC_MODELING",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (startAddingEmptyInvestmentsLine) 1`] = `
Array [
  Object {
    "type": "START_ADDING_EMPTY_INVESTMENTS_LINE",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (startAddingQOps) 1`] = `
Array [
  Object {
    "chipsNumber": "chipX",
    "type": "START_ADDING_Q_OPS",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (startApplyingProfitLoss) 1`] = `
Array [
  Object {
    "type": "START_APPLYING_PROFIT_LOSS",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (startBuyingWire) 1`] = `
Array [
  Object {
    "type": "START_BUYING_WIRE",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (startCountingRisk) 1`] = `
Array [
  Object {
    "type": "START_COUNTING_RISK",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (startDecreasingOperations) 1`] = `
Array [
  Object {
    "type": "START_DECREASING_OPERATIONS",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (startDecreasingOps) 1`] = `
Array [
  Object {
    "type": "START_DECREASING_OPS",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (startImprovingMarketing) 1`] = `
Array [
  Object {
    "act": Object {
      "a": 5,
    },
    "type": "START_IMPROVING_MARKETING",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (startInvestmentsDeposit) 1`] = `
Array [
  Object {
    "type": "START_INVESTMENTS_DEPOSIT",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (startInvestmentsWithdraw) 1`] = `
Array [
  Object {
    "type": "START_INVESTMENTS_WITHDRAW",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (startMegaClippers) 1`] = `
Array [
  Object {
    "type": "START_MEGACLIPPERS",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (startNewTournament) 1`] = `
Array [
  Object {
    "type": "START_NEW_TOURNAMENT",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (startRunningStrategicModeling) 1`] = `
Array [
  Object {
    "type": "START_RUNNING_STRATEGIC_MODELING",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (startSubtractingQOps) 1`] = `
Array [
  Object {
    "chipsNumber": "chipX",
    "type": "START_SUBTRACTING_Q_OPS",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (startTimer) 1`] = `
Array [
  Object {
    "type": "START_TIMER",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (startUpdatingInvestmentLines) 1`] = `
Array [
  Object {
    "type": "START_UPDATING_INVESTMENTS_LINE",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (startUpdatingScreen) 1`] = `
Array [
  Object {
    "type": "START_UPDATING_SCREEN",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (startUpdatingUnsoldInventory) 1`] = `
Array [
  Object {
    "type": "START_UPDATING_UNSOLD_INVENTORY",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (stop) 1`] = `
Array [
  Object {
    "type": "STOP",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (stopAddingEmptyInvestmentsLine) 1`] = `
Array [
  Object {
    "type": "STOP_ADDING_EMPTY_INVESTMENTS_LINE",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (stopAddingQOps) 1`] = `
Array [
  Object {
    "type": "STOP_ADDING_Q_OPS",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (stopDecreasingOps) 1`] = `
Array [
  Object {
    "type": "STOP_DECREASING_OPERATIONS",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (stopSubtractingQOps) 1`] = `
Array [
  Object {
    "type": "STOP_SUBTRACTING_Q_OPS",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (stopTournament) 1`] = `
Array [
  Object {
    "type": "STOP_TOURNAMENT",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (stopUpdatingInvestmentLines) 1`] = `
Array [
  Object {
    "type": "STOP_UPDATING_INVESTMENTS_LINE",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (stopUpdatingScreen) 1`] = `
Array [
  Object {
    "type": "STOP_UPDATING_SCREEN",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (stopUpdatingUnsoldInventory) 1`] = `
Array [
  Object {
    "type": "STOP_UPDATING_UNSOLD_INVENTORY",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (strategyChosen) 1`] = `
Array [
  Object {
    "strategy": "RANDOM",
    "type": "STRATEGY_CHOSEN",
    "val": true,
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (subtractQOps) 1`] = `
Array [
  Object {
    "chipsNumber": "chipX",
    "type": "SUBTRACT_Q_OPS",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (switchOffOrOnAutoAndMegaClippers) 1`] = `
Array [
  Object {
    "type": "SWITCH_OFF_OR_ON_AUTO_AND_MEGA_CLIPPERS",
    "val": true,
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (toggleAutoClippersButton) 1`] = `
Array [
  Object {
    "type": "TOGGLE_AUTO_CLIPPERS_BUTTON",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (toggleAutoWireBuyer) 1`] = `
Array [
  Object {
    "type": "TOGGLE_AUTO_WIRE_BUYER",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (toggleChip) 1`] = `
Array [
  Object {
    "chipsNumber": "chipX",
    "type": "TOGGLE_CHIP",
    "val": true,
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (toggleDropdownInvestments) 1`] = `
Array [
  Object {
    "type": "TOGGLE_DROPDOWN_INVESTMENTS",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (toggleDropdownStrategicModeling) 1`] = `
Array [
  Object {
    "type": "TOGGLE_DROPDOWN_STRATEGIC_MODELING",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (toggleMakePaperclipButton) 1`] = `
Array [
  Object {
    "type": "TOGGLE_MAKE_PAPERCLIP_BUTTON",
    "val": true,
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (toggleMarketingButton) 1`] = `
Array [
  Object {
    "type": "TOGGLE_MARKETING_BUTTON",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (toggleMegaClippersButton) 1`] = `
Array [
  Object {
    "type": "TOGGLE_MEGA_CLIPPERS_BUTTON",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (toggleNewTournamentButton) 1`] = `
Array [
  Object {
    "type": "TOGGLE_NEW_TOURNAMENT_BUTTON",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (toggleWireButton) 1`] = `
Array [
  Object {
    "type": "TOGGLE_WIRE_BUTTON",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (toggleWireBuyerProject) 1`] = `
Array [
  Object {
    "type": "TOGGLE_WIRE_BUYER_PROJECT",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (tournamentDuration) 1`] = `
Array [
  Object {
    "type": "TOURNAMENT_DURATION",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (tournamentStarted) 1`] = `
Array [
  Object {
    "type": "TOURNAMENT_STARTED",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (tournamentState) 1`] = `
Array [
  Object {
    "type": "TOURNAMENT_STATE",
    "val": true,
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (trustPlusOne) 1`] = `
Array [
  Object {
    "type": "TRUST_PLUS_ONE",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (trustPlusOneFromProject) 1`] = `
Array [
  Object {
    "type": "TRUST_PLUS_ONE_FROM_PROJECT",
    "val": 3,
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (updateAvgClipsSoldPerSec) 1`] = `
Array [
  Object {
    "type": "UPDATE_AVG_CLIPS_SOLD_PER_SEC",
    "val": 7,
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (updateAvgRevPerSec) 1`] = `
Array [
  Object {
    "type": "UPDATE_AVG_REV_PER_SEC",
    "val": 5,
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (updateClipsPerSec) 1`] = `
Array [
  Object {
    "type": "UPDATE_CLIPS_PER_SEC",
    "val": 7,
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (updateFakeInvestmentsCash) 1`] = `
Array [
  Object {
    "cash": 552,
    "type": "UPDATE_FAKE_INVESTMENTS_CASH",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (updateFunds) 1`] = `
Array [
  Object {
    "type": "UPDATE_FUNDS",
    "value": 5,
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (updateFundsWithdraw) 1`] = `
Array [
  Object {
    "type": "UPDATE_FUNDS_WITHDRAW",
    "val": 3534,
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (updateInvestmentsCash) 1`] = `
Array [
  Object {
    "cash": 644,
    "type": "UPDATE_INVESTMENTS_CASH",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (updateInvestmentsDelay) 1`] = `
Array [
  Object {
    "delayLines": 3000,
    "delayScreen": 4000,
    "type": "UPDATE_INVESTMENTS_DELAY",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (updateInvestmentsLines) 1`] = `
Array [
  Object {
    "array": Array [
      Object {
        "a": 4,
      },
      Object {
        "a": 7,
      },
    ],
    "type": "UPDATE_INVESTMENTS_LINES",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (updateInvestmentsStocks) 1`] = `
Array [
  Object {
    "stocks": 745,
    "type": "UPDATE_INVESTMENTS_STOCKS",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (updateInvestmentsTotal) 1`] = `
Array [
  Object {
    "total": 543,
    "type": "UPDATE_INVESTMENTS_TOTAL",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (updateMaxPublicDemand) 1`] = `
Array [
  Object {
    "type": "UPDATE_MAX_PUBLIC_DEMAND",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (updateNewTournamentCost) 1`] = `
Array [
  Object {
    "type": "UPDATE_NEW_TOURNAMENT_COST",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (updateOps) 1`] = `
Array [
  Object {
    "type": "UPDATE_OPS",
    "val": 34,
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (updatePublicDemand) 1`] = `
Array [
  Object {
    "type": "UPDATE_PUBLIC_DEMAND",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (updateStrategicModelingCurrentList) 1`] = `
Array [
  Object {
    "obj": Object {
      "a": 5,
      "b": 8,
    },
    "round": 9,
    "type": "UPDATE_STRATEGIC_MODELING_CURRENT_LIST",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (updateStrategicModelingData) 1`] = `
Array [
  Object {
    "obj": Object {
      "a": 5,
      "b": 5,
    },
    "type": "UPDATE_STRATEGIC_MODELING_DATA",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (updateStrategicModelingRound) 1`] = `
Array [
  Object {
    "round": 4,
    "type": "UPDATE_STRATEGIC_MODELING_ROUND",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (updateUnsoldInventory) 1`] = `
Array [
  Object {
    "type": "UPDATE_UNSOLD_INVENTORY",
  },
]
`;

exports[`businessActions Dispatches the correct action and payload (wireExists) 1`] = `
Array [
  Object {
    "type": "WIRE_EXISTS",
    "val": true,
  },
]
`;
